src/engine/source/cmsync/src/cm.cpp

```diff
@@ -0,0 +1,618 @@
+#include <cmsync/cmsync.hpp>
+
+#include <base/logging.hpp>
+
+#include "coreoutput.hpp"
+
+// Helper functions and definitions
+namespace
+{
+const base::Name G_POLICY_NAME {"policy/wazuh/0"};     ///< Default policy name
+const base::Name G_FILTER_NAME {"filter/allow-all/0"}; ///< Filter that allows all events
+
+/**
+ * @brief Get a shared pointer from a weak pointer or throw an exception if the pointer is expired
+ * @tparam T Type of the pointer
+ * @param weakPtr Weak pointer to get the shared pointer from
+ * @param name Name of the instance to include in the exception message
+ * @return const auto Shared pointer
+ */
+template<typename T>
+auto getHandlerOrThrow(const std::weak_ptr<T>& weakPtr, std::string_view name)
+{
+    const auto ptr = weakPtr.lock();
+    if (!ptr)
+    {
+        throw std::runtime_error(fmt::format("{} instance is no longer available", name));
+    }
+    return ptr;
+}
+
+/**
+ * @brief Get a filter that allows all events
+ * @return json::Json JSON representation of the filter
+ */
+json::Json getAllowAllFilter()
+{
+    // TODO: Move "/name" to common header
+    json::Json filter {};
+    filter.setString(G_FILTER_NAME.fullName(), "/name");
+    return filter;
+}
+
+} // namespace
+
+namespace cm::sync
+{
+
+/************************************************************************************
+ * ICMSync interface implementation
+ ************************************************************************************/
+void CMSync::deploy(const std::shared_ptr<cti::store::ICMReader>& ctiStore)
+{
+    namespace acns = api::catalog;
+
+    // Load from files and validate the core outputs
+    wazuhCoreOutput(true);
+
+    // TODO: We need validate the new ruleset before remove the old ones
+
+    // Remove all decoders and integrations from the catalog.
+    cleanCatalog(m_ctiNS, {acns::Resource::Type::decoder, acns::Resource::Type::integration});
```

Comment:
If there's a failure in any of this deletes we might have problems on the next steps